packages/url-launcher/android/src/main/java/io/flutter/plugins/url_launcher/UrlLauncherPlugin.java

```diff
@@ -13,34 +18,51 @@
  * UrlLauncherPlugin
  */
 public class UrlLauncherPlugin implements MethodCallHandler {
-  private FlutterActivity activity;
-
-  public static UrlLauncherPlugin register(FlutterActivity activity) {
-    return new UrlLauncherPlugin(activity);
-  }
-
-  private UrlLauncherPlugin(FlutterActivity activity) {
-    this.activity = activity;
-    new MethodChannel(
-            activity.getFlutterView(), "plugins.flutter.io/URLLauncher").setMethodCallHandler(this);
-  }
-
-  @Override
-  public void onMethodCall(MethodCall call, Result result) {
-    if (call.method.equals("UrlLauncher.launch")) {
-      launchURL((String) call.arguments);
-      result.success(null);
-    } else {
-      result.notImplemented();
+    private FlutterActivity activity;
+
+    public static UrlLauncherPlugin register(FlutterActivity activity) {
+        return new UrlLauncherPlugin(activity);
     }
-  }
-  private void launchURL(String url) {
-    try {
-      Intent launchIntent = new Intent(Intent.ACTION_VIEW);
-      launchIntent.setData(Uri.parse(url));
-      activity.startActivity(launchIntent);
-    } catch (java.lang.Exception exception) {
-      // Ignore parsing or ActivityNotFound errors
+
+    private UrlLauncherPlugin(FlutterActivity activity) {
+        this.activity = activity;
+        new MethodChannel(
+                activity.getFlutterView(), "plugins.flutter.io/URLLauncher").setMethodCallHandler(this);
+    }
+
+    @Override
+    public void onMethodCall(MethodCall call, Result result) {
+        String url = call.arguments();
+        if (call.method.equals("UrlLauncher.canLaunch")) {
+            canLaunch(url, result);
+        } else if (call.method.equals("UrlLauncher.launch")) {
+            launchURL(url, result);
+        } else {
+            result.notImplemented();
+        }
+    }
+
+    private void launchURL(String url, Result result) {
+        try {
+            Intent launchIntent = new Intent(Intent.ACTION_VIEW);
+            launchIntent.setData(Uri.parse(url));
+            activity.startActivity(launchIntent);
+            result.success(null);
+        } catch (java.lang.Exception exception) {
```

Comment:
No need for the `java.lang` package name. It's implicit.

Does the code require you to catch `Exception`? The `onMethodCall` is called within a try-catch block by the framework, so any uncaught `RuntimeException`s are caught there, logged, and error results are sent back to the Dart side.